src/types/TWAPWithIncreasingMinPartLimit.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {ComposableCoW} from "../ComposableCoW.sol";
+
+import {
+    IERC20,
+    IConditionalOrder,
+    IConditionalOrderGenerator,
+    GPv2Order,
+    BaseConditionalOrder
+} from "../BaseConditionalOrder.sol";
+import {TWAPOrder} from "./twap/libraries/TWAPOrder.sol";
+
+// --- error strings
+
+/// @dev The order is not within the TWAP bundle's span.
+string constant NOT_WITHIN_SPAN = "not within span";
+/// @dev The order count is not > 1.
+string constant INVALID_ORDER_COUNT = "invalid order count";
+/// @dev The part id is not > 0 and <= n or total number of parts.
+string constant PART_NUMBER_OUT_OF_RANGE = "part number out of range";
+/// @dev Invalid number of total number of parts, i.e., not greater than 1 and <= type(uint32).max.
+string constant INVALID_NUM_PARTS = "invalid num parts";
+
+contract TWAPWithIncreasingMinPartLimit is BaseConditionalOrder {
+    // --- types
+
+    struct Data {
+        IERC20 sellToken; // The token being sold
+        IERC20 buyToken; // The token being bought
+        address receiver; // The order owner (user who initiated the order)
+        uint256 partSellAmount; // The amount to sell in each part
+        uint256 startPrice; // Sell price for the first part
+        uint256 endPrice; // Sell price for the last part
+        uint256 t0; // Start timestamp
+        uint256 t; // Time interval for each part
```

Comment:
Reading the suggested order type (from the tweet you linked), I don't really understand how time is coming into play here? I understand the user wants to place n orders with one signature to buy at different price points (e.g. buy 1% at $1, 1% at $1.01, 1% at $1.02 ... $2).

If the price moves, all orders in the price range should be executed (there shouldn't be any notion of start time)

Reply:
Remove `startTime` and `endPrice` and updated the logic to get the percentage increase from off-chain and compute the `minPartLimit ` for each part.